pyproject.toml

```diff
@@ -0,0 +1,9 @@
+[tool.codespell]
+# note: pre-commit passes explicit lists of files here, which this skip file list doesn't override -
+# this is only to allow you to run codespell interactively
+skip = "./.git,./.github,./cpp/build,.*egg-info.*,./.mypy_cache,.*_skbuild,CHANGELOG.md,_stop_words.py,,*stemmer.*"
+# ignore short words, and typename parameters like OffsetT
+ignore-regex = "\\b(.{1,4}|[A-Z]\\w*T)\\b"
+ignore-words-list = "inout,numer,startd,couldn"
+builtin = "clear"
+quiet-level = 3
```

Comment:
It's not immediately obvious to me what these config options do, maybe you can add comments there as well?